apps/src/netsim/netsimUtils.js

```diff
@@ -37,12 +38,72 @@ exports.jQuerySvgElement = function (type) {
     var oldClasses = newElement.attr('class');
     if (!oldClasses) {
       newElement.attr('class', className);
-    } else if (!oldClasses.split(/\s+/g).some(function (existingClass) {
-          return existingClass === className;
-        })) {
+    } else if (!newElement.hasClass(className)) {
       newElement.attr('class', oldClasses + ' ' + className);
     }
-    // Return element for chaining
+    return newElement;
+  };
+
+  /**
+   * Override removeClass since jQuery removeClass doesn't work on svg.
+   * Removes the given classname if it exists on the element.
+   * @param {string} className
+   * @returns {jQuery} for chaining
+   */
+  newElement.removeClass = function (className) {
+    var oldClasses = newElement.attr('class');
+    if (oldClasses) {
+      var newClasses = oldClasses
+          .split(/\s+/g)
+          .filter(function (word) {
+            return word !== className;
+          })
+          .join(' ');
+      newElement.attr('class', newClasses);
+    }
+    return newElement;
+  };
+
+  /**
+   * Override hasClass since jQuery hasClass doesn't work on svg.
+   * Checks whether the element has the given class.
+   * @param {string} className
+   * @returns {boolean}
+   */
+  newElement.hasClass = function (className) {
+    var oldClasses = newElement.attr('class');
+    return oldClasses && oldClasses.split(/\s+/g)
+        .some(function (existingClass) {
+          return existingClass === className;
+        });
+  };
+
+  /**
+   * Override toggleClass since jQuery toggleClass doesn't work on svg.
+   *
+   * Two versions:
+   *
+   * toggleClass(className) reverses the state of the class on the element;
+   *   if it has the class it gets removed, if it doesn't have the class it
+   *   gets added.
+   *
+   * toggleClass(className, shouldHaveClass) adds or removes the class on the
+   *   element depending on the value of the second argument.
+   *
+   *
+   * @param {string} className
+   * @param {boolean} [shouldHaveClass]
+   * @returns {jQuery} for chaining
+   */
+  newElement.toggleClass = function (className, shouldHaveClass) {
+    // Default second argument - if not provided, we flip the current state
+    shouldHaveClass = utils.valueOr(shouldHaveClass, !newElement.hasClass(className));
+
+    if (shouldHaveClass) {
+      newElement.addClass(className);
+    } else {
+      newElement.removeClass(className);
+    }
```

Comment:
This all looks okay to me, but I wonder if there's an existing jQuery plugin we could use that has run into/fixed any cross browser issues.